miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/StorageLocationRestController.java

```diff
@@ -110,32 +114,34 @@ private StorageLocation getFreezer(long id) {
 
   @PostMapping(value = "/freezers/{id}/stacks")
   public @ResponseBody StorageLocationDto addFreezerStack(@PathVariable(name = "id", required = true) long id,
-      @RequestParam(name = "height", required = true) int height) {
+      @RequestParam(name = "height", required = true) int height,
+      @RequestParam(name = "identificationBarcode", required = false) String barcode) {
     StorageLocation freezer = getFreezer(id);
     if (height < 1) {
       throw new RestException("Invalid stack height", Status.BAD_REQUEST);
     }
 
-    return createStack(freezer, height);
+    return createStack(freezer, height, barcode);
   }
 
   @PostMapping(value = "/rooms")
   @ResponseStatus(HttpStatus.CREATED)
   @ResponseBody
   public StorageLocationDto addRoom(@RequestParam(name = "alias", required = true) String alias,
       @RequestParam(name = "identificationBarcode", required = false) String identificationBarcode) throws IOException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'java.io.IOException', as it cannot be thrown from method's body. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)